node/node.go

```diff
@@ -194,22 +195,27 @@ func activateShhService(stack *node.Node, config *params.NodeConfig, db *leveldb
 		logger.Info("SHH protocol is disabled")
 		return nil
 	}
-	if err := stack.Register(func(*node.ServiceContext) (node.Service, error) {
-		return timesource.Default(), nil
-	}); err != nil {
-		return err
+	if config.WhisperConfig.EnableNTPSync {
+		if err := stack.Register(func(*node.ServiceContext) (node.Service, error) {
+			return timesource.Default(), nil
+		}); err != nil {
+			return err
+		}
 	}
 
 	err = stack.Register(func(ctx *node.ServiceContext) (node.Service, error) {
-		var timeSource *timesource.NTPTimeSource
-		if err := ctx.Service(&timeSource); err != nil {
-			return nil, err
-		}
 		whisperServiceConfig := &whisper.Config{
 			MaxMessageSize:     whisper.DefaultMaxMessageSize,
 			MinimumAcceptedPOW: 0.001,
-			TimeSource:         timeSource.Now,
 		}
+
+		whisperServiceConfig.TimeSource = time.Now
```

Comment:
Why not keep this assignment in the initialization block? Is it to have both assignments to `TimeSource` near each other?